### PR TITLE
delete deprecated buffer props

### PR DESCRIPTION
**Pull request by xintongxia**
#### Background

buffer bug

#### Change List
- base-attribute

### CODE REVIEW

File: modules/core/src/lib/base-attribute.js [hunk 1]
```diff
@@ -127,6 +127,12 @@ export default class BaseAttribute {
       props.accessor.size = props.size;
     }
     delete props.size;
+    if ('type' in props) {
+      delete props.type;
```
Comment (ibgreen) [inline modules/core/src/lib/base-attribute.js]: Should we put it in `props.accessor` before we delete it?

**Comment by 1chandu** [comment]
My understanding was these props were sent by the application/layer, but if these are internally set by `AttributeManager`, we need a more thorough examination of the the usage before deleting it.

File: modules/core/src/lib/base-attribute.js (hunk 1 above)
Comment (Pessimistress) [reply modules/core/src/lib/base-attribute.js]: Can we just avoid merging `opts` with `props`? I do not believe the buffer's accessor is used anywhere (the `Attribute` class stores its own accessor) so there's no point setting it.

File: modules/core/src/lib/base-attribute.js [hunk 2]
```diff
@@ -110,15 +110,15 @@ export default class BaseAttribute {
...
-    const props = Object.assign({}, opts, {
+    const props = {
       id: this.id,
       target: this.target,
       accessor: {
         type: this.type
       }
-    });
+    };
     if (Number.isFinite(props.divisor)) {
```
Comment (Pessimistress) [inline modules/core/src/lib/base-attribute.js]: Remove

Comment (xintongxia) [reply modules/core/src/lib/base-attribute.js]: remove passing `divisor` and `size` to `Buffer` constructor?

Comment (Pessimistress) [reply modules/core/src/lib/base-attribute.js]: You are assigning them to `props.accessor` which we are no longer passing.

**Comment by Pessimistress** [comment]
As discussed offline, this should not be merged before we cut the 7.3-release branch.

**Pull request merged**